### PR TITLE
Cleaned up cryptography code snippets for Key Vault Keys.

### PR DESCRIPTION
**Pull request by vcolin7**
Fixes: #18205.

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClientJavaDocCodeSnippets.java [hunk 1]
```diff
@@ -102,10 +104,11 @@ public CryptographyAsyncClient createAsyncClientWithPipeline() {
      */
     public void getKeySnippets() {
         CryptographyAsyncClient cryptographyAsyncClient = createAsyncClient();
+
         // BEGIN: com.azure.security.keyvault.keys.cryptography.CryptographyAsyncClient.getKey
         cryptographyAsyncClient.getKey()
-            .subscriberContext(reactor.util.context.Context.of(key1, value1, key2, value2))
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClientJavaDocCodeSnippets.java:107]: why are we removing subscriber context ?

File: sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClientJavaDocCodeSnippets.java [hunk 2]
```diff
@@ -261,31 +265,25 @@ public void wrapKeyUnwrapKey() {
...
         // END: com.azure.security.keyvault.keys.cryptography.CryptographyAsyncClient.signData#SignatureAlgorithm-byte
 
         // BEGIN: com.azure.security.keyvault.keys.cryptography.CryptographyAsyncClient.verifyData#SignatureAlgorithm-byte-byte
-        cryptographyAsyncClient.verify(SignatureAlgorithm.ES256, data, signature)
-            .subscriberContext(reactor.util.context.Context.of(key1, value1, key2, value2))
+        byte[] myData = new byte[100];
+        new Random(0x1234567L).nextBytes(myData);
+        byte[] signature = new byte[100];
+
+        cryptographyAsyncClient.verify(SignatureAlgorithm.ES256, myData, signature)
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClientJavaDocCodeSnippets.java:284]: same, code won't work, with above input,

File: sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientJavaDocCodeSnippets.java [hunk 3]
```diff
@@ -186,37 +185,59 @@ public void decrypt() {
...
 
         // BEGIN: com.azure.security.keyvault.keys.cryptography.CryptographyClient.verify#SignatureAlgorithm-byte-byte
-        VerifyResult verifyResult = cryptographyClient.verify(SignatureAlgorithm.ES256, digest, signature);
-        System.out.printf("Verification status %s", verifyResult.isValid());
+        byte[] myData = new byte[100];
+        new Random(0x1234567L).nextBytes(myData);
+        MessageDigest messageDigest = MessageDigest.getInstance("SHA-256");
+        messageDigest.update(myData);
+        byte[] myDigest = messageDigest.digest();
+        byte[] signature = new byte[100];
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientJavaDocCodeSnippets.java:222]: code won't work, either build the signature first or leave it out of the scope with a comment prompting user to sign firat.

File: sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientJavaDocCodeSnippets.java [hunk 4]
```diff
@@ -186,37 +185,59 @@ public void decrypt() {
...
 
         // BEGIN: com.azure.security.keyvault.keys.cryptography.CryptographyClient.verify#SignatureAlgorithm-byte-byte-Context
-        VerifyResult verifyResponse = cryptographyClient.verify(SignatureAlgorithm.ES256, digest, signature);
-        System.out.printf("Verification status %s", verifyResponse.isValid(), new Context(key2, value2));
+        byte[] dataBytes = new byte[100];
+        new Random(0x1234567L).nextBytes(dataBytes);
+        MessageDigest msgDigest = MessageDigest.getInstance("SHA-256");
+        msgDigest.update(dataBytes);
+        byte[] digestBytes = msgDigest.digest();
+        byte[] signatureBytes = new byte[100];
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientJavaDocCodeSnippets.java:235]: same as above.

File: sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientJavaDocCodeSnippets.java [hunk 5]
```diff
@@ -283,30 +305,46 @@ public void wrapKeyUnwrapKey() {
...
+
+        System.out.printf("Verification status: %s.\n", verifyResult.isValid());
         // END: com.azure.security.keyvault.keys.cryptography.CryptographyClient.verifyData#SignatureAlgorithm-byte-byte
 
         // BEGIN: com.azure.security.keyvault.keys.cryptography.CryptographyClient.verifyData#SignatureAlgorithm-byte-byte-Context
-        VerifyResult verifyResponse = cryptographyClient.verify(SignatureAlgorithm.ES256, data, signature);
-        System.out.printf("Verification status %s", verifyResponse.isValid(), new Context(key2, value2));
+        byte[] dataToVerify = new byte[100];
+        new Random(0x1234567L).nextBytes(dataToVerify);
+        byte[] mySignature = new byte[100];
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientJavaDocCodeSnippets.java:342]: same.

File: sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientJavaDocCodeSnippets.java [hunk 6]
```diff
@@ -283,30 +305,46 @@ public void wrapKeyUnwrapKey() {
...
         // END: com.azure.security.keyvault.keys.cryptography.CryptographyClient.signData#SignatureAlgorithm-byte-Context
 
         // BEGIN: com.azure.security.keyvault.keys.cryptography.CryptographyClient.verifyData#SignatureAlgorithm-byte-byte
-        VerifyResult verifyResult = cryptographyClient.verify(SignatureAlgorithm.ES256, data, signature);
-        System.out.printf("Verification status %s", verifyResult.isValid());
+        byte[] myData = new byte[100];
+        new Random(0x1234567L).nextBytes(myData);
+        byte[] signature = new byte[100];
+
+        VerifyResult verifyResult = cryptographyClient.verify(SignatureAlgorithm.ES256, myData, signature);
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientJavaDocCodeSnippets.java:334]: same.

File: sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClientJavaDocCodeSnippets.java (hunk 1 above)
Comment (srnagar) [inline sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClientJavaDocCodeSnippets.java:107]: Use `contextWrite()` instead.
https://projectreactor.io/docs/core/release/reference/#context

Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClientJavaDocCodeSnippets.java:107]: Should all async samples include the context?

Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClientJavaDocCodeSnippets.java:107]: Should all async samples include the context?

**Comment by vcolin7** [comment]
Added back the context using @srnagar's recommendation and made the changes suggested by @g2vinay.

**Review by g2vinay — APPROVED** [review]

**Pull request merged**